salt/metalk8s/orchestrate/deploy_node.sls

```diff
@@ -113,6 +113,21 @@ Wait minion available:
 
 {%- endif %}
 
+# Prepare etcd and dependencies
+{%- if 'etcd' in pillar.get('metalk8s', {}).get('nodes', {}).get(node_name, {}).get('roles', []) %}
+
+Prepare etcd dependencies:
+  salt.state:
+    - tgt: {{ node_name }}
+    - sls: metalk8s.roles.etcd.prepared
```

Comment:
You can't run this state on the minion as this is something that need `etcd3` python lib which is only available on the salt-master